Prototyping/Sit/sit.js

```diff
@@ -0,0 +1,692 @@
+                        url: OVERLAY_URL_STANDUP,
+                        ignoreRayIntersection: false,
+                        drawInFront: true,
+                        visible: true,
+                        emissive: true
+                    });
+                }
+            },
+            remove: function () {
+                if (overlayStandUp) {
+                    Overlays.deleteOverlay(overlayStandUp);
+                    overlayStandUp = null;
+                }
+            },
+            update: function () {
+                if (overlayStandUp) {
+                    Overlays.editOverlay(overlayStandUp, { // too inefficient
+                        rotation: Camera.orientation,
+                        position: Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: 0, z: -1 }))
+                    });
+                }
+            }
+        },
+
+        showSittable: function () { // Show overlays when I'm close to the seat
+
+            if (isInEditMode()) {
+                return;
+            }
+
+            if (overlaySittable === null) { // Make an overlay if there isn't one
+                if (this.sittable.shouldShow()) {
+                    this.sittable.create();
+                    if (overlayIntervalTransparency === null) {
+                        this.sittable.lerpTransparency();
+                    }
+                }
+            } else if (!utils.canSitDesktop()) {
+                this.sittable.remove();
+            }
+        }
+    };
+
+    var utils = {
+
+        // Is the seat used
+        checkSeatForAvatar: function () {
+            return deadlockGate;
+
+            // var isOccupied = false;
+
+            // if(deadlockGate) {
+            // return true;
+            // }
+
+            // var nearbyAvatars = AvatarList.getAvatarsInRange(seatCenterPosition, SITTING_SEARCH_RADIUS);
+
+            // if (nearbyAvatars.length > 0) {
+            // isOccupied = true;
+            // }
+
+            // if(deadlockGate){
+            // isOccupied = true;
+            // }
+
+            // return isOccupied;
+        },
+        rolesToOverride: function () {
+            return MyAvatar.getAnimationRoles().filter(function (role) {
+                return !(role.startsWith("right") || role.startsWith("left"));
+            });
+        },
+        canSitDesktop: function () {
+            if (!chairProperties) this.setChairProperties();
+            var distanceFromSeat = Vec3.distance(MyAvatar.position, chairProperties.position);
+            return distanceFromSeat < SITTABLE_DISTANCE_MAX && !this.checkSeatForAvatar();
+        },
+
+        calculatePinHipPosition: function () {
+            if (!chairProperties) this.setChairProperties();
+
+            // get chairProperties to calculate middle of chair
+            seatCenterPosition = chairProperties.position;
+
+            var yOffset = chairProperties.dimensions.y / 2 * chairOffsetRatio;
```

Comment:
This chairOffsetRatio maybe should be multiplied by some factor depending on the MyAvatar.getHeight().  so that tall avatars will have a large offset and short avatar would have a smaller offset.  For example:
```
var DEFAULT_AVATAR_HEIGHT = 1.755;
var chairOffsetRatio = 0.2 * (MyAvatar.getHeight() / DEFAULT_AVATAR_HEIGHT);
```

Reply:
Flagging this comment to come back to.